### ISSUE TITLE
GH3 PC, RPCS3, and YARG having incorrect inputs and behavior. XBOX Controller

### ISSUE THREAD

**Issue by red6332** [TheNathannator/RB4InstrumentMapper#39]
So oddly enough YARG, RPCS3, and GH3(I assume World Tour as well) have this weird input issue. You can navigate the menus perfectly fine, but once a song starts the inputs arent working properly. For example pressing the green button activates the orange gem. Some buttons will either activate no gem or even multiples, such as blue activating green and yellow. Im having a hard time remembering if the PC version of 3 does this, but when you press any button, it acts like as if you are using a regular controller. Meaning pressing a button both strums and activates the wrong gems. I have followed various guides for RPCS3 including the RPCS3 site as well. As for YARG I just dont understand it it jist does the same thing even when I map the buttons properly. Nothing seems to work and Im still trying to look into it, but I dont have programming or technical knowledge on the hardware. Oddly enough I do not have any of these issues on Clone Hero, other than clone hero not detecting my controller from time to time.

**Comment by TheNathannator** [comment]
The reason GH3 PC doesn't work correctly with this is that RB4InstrumentMapper can only emulate Xbox 360 gamepads, it can't mark them as guitars unfortunately. You'll need the older version of x360ce that hooks  XInput calls to fix this.

For RPCS3 and YARG, this is a configuration issue on your end somewhere:
- RPCS3 requires you to set the controller type in the configuration menu. Whether you set it as a Rock Band guitar or a Guitar Hero guitar depends on what game you're playing.
- YARG has separate navigation and gameplay controls, make sure you configure both. It also has a dialog that shows up the first time you add a gamepad, which allows you to quickly configure for WiitarThing or RB4IM usage. To get it to show up again, hit Remove Device and flip the toggle that offers to reset binds to default before clicking on the device in the list, then re-add it.

**Comment by red6332** [comment]
Oh shit I didnt know you can send me an email directly. Yeah the RPCS3 thing Ive done 2 actual walkthroughs 4 times and it operates the same. I just learned about X360ce. Which version would you recommend? Also thanks for the reply.

Sent via the Samsung Galaxy S24 Ultra, an AT&T 5G smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by TheNathannator** [comment]
Yeah, GitHub sends you emails when someone replies, and you can reply directly in the email and it shows up here.

You want the latest v3 version of x360ce, v4 uses ViGEmBus like RB4IM does and will suffer from the same issue of only being able to emulate gamepads. I'm not familiar with the exact setup and usage process though, so I won't be of much help beyond this lol

**Comment by red6332** [comment]
Thank you for the help, I really appreciate it especially since you gave me info beyond what others have told me. Lol you have yourself a great day and night!

Sent via the Samsung Galaxy S24 Ultra, an AT&T 5G smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by red6332** [comment]
Im sorry to bother you, but is x360ce supposed to just detect the controller? Because I need to use RB4I if I want the app to detect it.

Sent via the Samsung Galaxy S24 Ultra, an AT&T 5G smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by TheNathannator** [comment]
You still need RB4InstrumentMapper running; you use x360ce to do another layer of remapping and configuration to make the game think the emulated gamepad is a guitar instead. There should be a subtype dropdown somewhere in the configuration tool.

Since x360ce overrides all of the XInput calls, the game will only see controllers which you've configured to be visible, so you need to make sure to configure things there.
